src/components/sync-dialog.tsx

```diff
@@ -0,0 +1,288 @@
+				),
+			},
+			sandbox: {
+				title: __( 'Pull from Sandbox' ),
+				description: __(
+					"Pulling will replace your Studio site's files and database with a copy from your sandbox site."
+				),
+			},
+			production: {
+				title: __( 'Pull from Production' ),
+				description: __(
+					"Pulling will replace your Studio site's files and database with a copy from your production site."
+				),
+			},
+			fromLabel: __( 'Pull' ),
+			toLabel: __( 'To' ),
+			subtitleSelector: __( 'What would you like to pull?' ),
+			envSync: __( 'Read more about <a>environment pull <ArrowIcon /></a>' ),
+			submit: __( 'Pull' ),
+		};
+	} else {
+		return {
+			staging: {
+				title: __( 'Push to Staging' ),
+				description: __(
+					'Pushing will replace the existing files and database with a copy from your local site.\n\n The staging site will be backed-up before any changes are applied.'
+				),
+			},
+			sandbox: {
+				title: __( 'Push to Sandbox' ),
+				description: __(
+					'Pushing will replace the existing files and database with a copy from your local site.\n\n The sandbox site will be backed-up before any changes are applied.'
+				),
+			},
+			production: {
+				title: __( 'Push to Production' ),
+				description: __(
+					'Pushing will replace the existing files and database with a copy from your local site.\n\n The production site will be backed-up before any changes are applied.'
+				),
+			},
+			fromLabel: __( 'Push' ),
+			toLabel: __( 'To' ),
+			subtitleSelector: __( 'What would you like to push?' ),
+			envSync: __( 'Read more about <a>environment push <ArrowIcon /></a>' ),
+			submit: __( 'Push' ),
+		};
+	}
+};
+
+const useEnvDetails = (
+	connectedSite: SyncSite
+): { label: string; envType: EnvironmentType; fillClass: string } => {
+	const { __ } = useI18n();
+
+	const envTypeValues = {
+		production: {
+			label: __( 'Production' ),
+			envType: 'production',
+			fillClass: 'fill-circle-env-production',
+		},
+		staging: {
+			label: __( 'Staging' ),
+			envType: 'staging',
+			fillClass: 'fill-circle-env-staging',
+		},
+		sandbox: {
+			label: __( 'Sandbox' ),
+			envType: 'sandbox',
+			fillClass: 'fill-sandbox-text',
+		},
+	} as const;
+
+	if ( connectedSite.isPressable ) {
+		return (
+			envTypeValues[ connectedSite.environmentType as EnvironmentType ] ?? envTypeValues.production
+		);
+	}
+
+	if ( connectedSite.isStaging ) {
+		return envTypeValues.staging;
+	} else {
+		return envTypeValues.production;
+	}
+};
+
+type SyncDialogProps = {
+	type: 'push' | 'pull';
+	localSite: SiteDetails;
+	remoteSite: SyncSite;
+	onSubmit: ( syncData: TreeNode[] ) => void;
+	onRequestClose: () => void;
+};
+
+export function SyncDialog( {
+	type,
+	localSite,
+	remoteSite,
+	onSubmit,
+	onRequestClose,
+}: SyncDialogProps ) {
+	const locale = useI18nLocale();
+	const copy = useCopy( type );
+
+	const [ showAllFiles, setShowAllFiles ] = useState( false );
+	const [ treeState, setTreeState ] = useState< TreeNode[] >( [
```

Comment:
Some of these labels are not translated.
I'm not sure if it makes sense to translate everything, but some I think should be, "Database" for example

Reply:
I think only Database should be translated, sin ce otehr folders are physical and not translated for customers on OS.
Translated Database, good catch 👍